File: clients/csharp/examples/collections-update-rename.cs
using DataStax.AstraDB.DataApi;
using DataStax.AstraDB.DataApi.Collections;
using DataStax.AstraDB.DataApi.Core;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing collection
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );
    var collection = database.GetCollection("**COLLECTION_NAME**");

    // Update a document
    var filterBuilder = Builders<Document>.CollectionFilter;
    var filter = filterBuilder.And(
      filterBuilder.Eq("title", "Into Shadows of Tomorrow"),
      filterBuilder.Eq("author", "Nicole Wright")
    );

    var update = Builders<Document>
      .CollectionUpdate.Rename("old_field", "new_field")
      .Rename("other_old_field", "other_new_field");

    await collection.UpdateOneAsync(filter, update);
  }
}

File: clients/typescript/examples/collections-update-rename.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

// Update a document
(async function () {
  const result = await collection.updateOne(
    {
      $and: [
        { title: "Into Shadows of Tomorrow" },
        { author: "Nicole Wright" },
      ],
    },
    { $rename: { old_field: "new_field", other_old_field: "other_new_field" } },
  );

  console.log(result);
})();
